src/Microsoft.AspNet.Mvc.TagHelpers/Internal/ModeMatchResult.cs

```diff
@@ -63,14 +64,13 @@ public class ModeMatchResult<TMode>
                     match => match.PresentAttributes.Any(
                         attribute => PartiallyMatchedAttributes.Contains(
                             attribute, StringComparer.OrdinalIgnoreCase)));
-
+                //TODO: What the heck is this thing?
                 logger.LogWarning(new PartialModeMatchLogValues<TMode>(uniqueId, viewPath, partialOnlyMatches));
```

Comment:
We still want to create the extension method here, just move `PartialModeMatchLogValues` to be a private class inside that class, and then use `LogWarning` as-is.